mantidproject/mantid
# Allow Preview Widgets to be Dockable

**Description of work.**
Make the widgets on the preview tab side-by-side and dockable. This means that they can be re-ordered on the gui or separated out and resized individually. 

**To test:**
1. Open the ISIS Reflectometry interface
2. Go to the Preview tab
3. Enter run `INTER45455_inst` and click Load (This data is available on this Epic and must be in your path: #31069)
4. Check that the widgets are nicely visible after loading
5. Check widgets can be undocked and still used
6. Set the angle >0
7. Draw a box on the inst viewer
8. Draw a box on the region selector
9. See the line plot update
10. Click apply
11. Check the chosen ROIs from the drawn boxes appear on the experiment settings tab. 

Fixes #34670 


*This does not require release notes* because **this tab is still work in progress so hasn't been shown to users yet.**




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
